pyscriptjs/src/main.ts

```diff
@@ -17,13 +17,24 @@ import { SplashscreenPlugin } from './plugins/splashscreen';
 import { ImportmapPlugin } from './plugins/importmap';
 import { StdioDirector as StdioDirector } from './plugins/stdiodirector';
 import type { PyProxy } from 'pyodide';
+import * as Synclink from 'synclink';
 // eslint-disable-next-line
 // @ts-ignore
 import pyscript from './python/pyscript/__init__.py';
 import { robustFetch } from './fetch';
+import { RemoteInterpreter } from './remote_interpreter';
 
 const logger = getLogger('pyscript/main');
 
+const old_val = Synclink.transferHandlers.get('throw').serialize;
+function new_val({ value }) {
+    const result = old_val({ value });
+    // @ts-ignore
+    result[0].value.$$isUserError = value.$$isUserError;
+    return result;
+}
+Synclink.transferHandlers.get('throw').serialize = new_val;
+
```

Comment:
Yeah it's monkey patching the error transfer handler to preserve the `$$isUserError` marker so that we can detect `UserError` subclasses in the error handling code.

Reply:
added a comment and renamed variables